exchange/utils.go

```diff
@@ -125,42 +132,86 @@ func splitImps(imps []openrtb.Imp) (map[string][]openrtb.Imp, []error) {
 
 	splitImps := make(map[string][]openrtb.Imp, len(imps))
 	var errList []error
+
 	for i := 0; i < len(imps); i++ {
-		thisImp := imps[i]
-		theseBidders := impExts[i]
-		for intendedBidder := range theseBidders {
-			if intendedBidder == "prebid" {
+		imp := imps[i]
+		impExt := impExts[i]
+
+		rawPrebidExt, ok := impExt["prebid"]
+
+		if ok {
+			var prebidExt openrtb_ext.ExtImpPrebid
+
+			if err := json.Unmarshal(rawPrebidExt, &prebidExt); err == nil && prebidExt.Bidder != nil {
+				if errs := sanitizedImpCopy(&imp, prebidExt.Bidder, rawPrebidExt, &splitImps); errs != nil {
+					errList = append(errList, errs...)
+				}
+
 				continue
 			}
+		}
 
-			otherImps, _ := splitImps[intendedBidder]
-			if impForBidder, err := sanitizedImpCopy(&thisImp, theseBidders, intendedBidder); err != nil {
-				errList = append(errList, err)
-			} else {
-				splitImps[intendedBidder] = append(otherImps, *impForBidder)
-			}
+		if errs := sanitizedImpCopy(&imp, impExt, rawPrebidExt, &splitImps); errs != nil {
+			errList = append(errList, errs...)
 		}
 	}
 
 	return splitImps, nil
 }
 
-// sanitizedImpCopy returns a copy of imp with its ext filtered so that only "prebid" and intendedBidder exist.
+// sanitizedImpCopy returns a copy of imp with its ext filtered so that only "prebid" and bidder params exist.
 // It will not mutate the input imp.
-// This function expects the "ext" argument to have been unmarshalled from "imp", so we don't have to repeat that work.
-func sanitizedImpCopy(imp *openrtb.Imp, ext map[string]json.RawMessage, intendedBidder string) (*openrtb.Imp, error) {
-	impCopy := *imp
-	newExt := make(map[string]json.RawMessage, 2)
-	if value, ok := ext["prebid"]; ok {
-		newExt["prebid"] = value
-	}
-	newExt["bidder"] = ext[intendedBidder]
-	extBytes, err := json.Marshal(newExt)
-	if err != nil {
-		return nil, err
+// This function will write the new imps to the output map passed in
+func sanitizedImpCopy(imp *openrtb.Imp,
+	bidderExts map[string]json.RawMessage,
+	rawPrebidExt json.RawMessage,
+	out *map[string][]openrtb.Imp) []error {
+
+	var prebidExt map[string]json.RawMessage
+	var errs []error
+
+	// We don't want to include other demand partners' bidder params
+	// in the sanitized imp
+	if err := json.Unmarshal(rawPrebidExt, &prebidExt); err == nil {
```

Comment:
I was going to ask if we should push this up to only do it once, but then I realized that the loop over bidders is now happening inside this function, rather than out in `cleanOpenRTBRequests()`